### PR TITLE
Offer

### PR DESCRIPTION
**Pull request by Sanjayvkp**

### CODE REVIEW

File: lib/core/constants/offer_constants/add_offer_page_constants.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class AddOfferPageConstants {
+  final String appbarTitle = 'Add Offers';
```
Comment (thadathilsinan) [inline lib/core/constants/offer_constants/add_offer_page_constants.dart]: prefix txt for the strng const

File: lib/core/constants/offer_constants/offer_page_constants.dart [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class OfferPageConstants {
+  final String offerText = 'Offer up to 30%';
+  final String appbarTitle = 'Offers';
```
Comment (thadathilsinan) [inline lib/core/constants/offer_constants/offer_page_constants.dart]: txt

File: lib/core/widgets/app_bar_widget.dart [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
-//   @override
-//   // TODO: implement preferredSize
-//   Size get preferredSize => throw UnimplementedError();
-
-//   @override
-//   // TODO: implement child
-//   Widget get child => throw UnimplementedError();
-// }
+  @override
+  Size get preferredSize => const Size.fromHeight(40);
```
Comment (thadathilsinan) [inline lib/core/widgets/app_bar_widget.dart]: use spaces

File: lib/core/widgets/elevated_add_button_widget.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      required this.textColor,
+      required this.onPressed,
+      required this.icon});
+
+  @override
+  Widget build(BuildContext context) {
+    OfferPageConstants constants = OfferPageConstants();
+    return SizedBox(
+      width: MediaQuery.sizeOf(context).width,
+      height: MediaQuery.sizeOf(context).height / 18,
```
Comment (thadathilsinan) [inline lib/core/widgets/elevated_add_button_widget.dart]: use spaces

File: lib/core/widgets/image_picker_widget.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+class ImagePickerWidget extends StatelessWidget {
+  const ImagePickerWidget({super.key});
+  @override
+  Widget build(BuildContext context) {
+    AppAssetsConstants iconConst = AppAssetsConstants();
+    return InkWell(
+      onTap: () {},
+      child: Container(
+        height: MediaQuery.sizeOf(context).height / 3.55,
```
Comment (thadathilsinan) [inline lib/core/widgets/image_picker_widget.dart]: use spaces

File: lib/features/offer/presentation/widgets/offer_banner_widget.dart [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          children: [
+            Padding(
+              padding: EdgeInsets.symmetric(horizontal: spaces.space_300),
+              child: Container(
+                height: spaces.space_500 * 5,
+                decoration: BoxDecoration(
+                    borderRadius: BorderRadius.circular(spaces.space_50),
+                    color: color.secondary,
+                    boxShadow: [
+                      BoxShadow(blurRadius: 4, color: color.textSubtle)
```
Comment (thadathilsinan) [inline lib/features/offer/presentation/widgets/offer_banner_widget.dart]: add into theme

File: lib/features/offer/presentation/widgets/textfield_widget.dart [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    final apptheme = AppTheme.of(context);
+
+    return TextField(
+      keyboardType: TextInputType.number,
+      controller: controller,
+      decoration: InputDecoration(
+          contentPadding: const EdgeInsets.all(0),
+          hintText: hintText,
+          hintStyle: apptheme.typography.h400.copyWith(
+            color: apptheme.colors.textSubtlest.withOpacity(.60),
```
Comment (thadathilsinan) [inline lib/features/offer/presentation/widgets/textfield_widget.dart]: check the color again

File: pubspec.lock
Comment (thadathilsinan) [inline pubspec.lock:1]: remove this file

**Review by thadathilsinan — CHANGES_REQUESTED** [review]

**Review by Sireen-Official — APPROVED** [review]
Review approved

**Review by thadathilsinan — APPROVED** [review]

**Comment by thadathilsinan** [comment]
Approved

**Pull request merged**